core/src/main/scala/org/broadinstitute/workbench/ccm/CostCalculator.scala

```diff
@@ -0,0 +1,93 @@
+package org.broadinstitute.workbench.ccm
+
+import java.time.{Duration, Instant}
+
+import cats.data.NonEmptyList
+import cats.implicits._
+import org.broadinstitute.workbench.ccm.pricing.{PriceList, PriceListKey, Prices, UsageType}
+
+
+object CostCalculator {
+
+  def getPriceOfCall(callMetaDataJson: MetadataResponse, priceList: PriceList): Either[Throwable, Double] = {
+    val ls: List[Either[NonEmptyList[String], Double]] = callMetaDataJson.calls.map { call =>
+      getPriceOfCall(call, priceList, Instant.now(), Instant.now()).leftMap(NonEmptyList.one)
+    }
+
+    ls.parSequence.leftMap(errors => new Exception(errors.toList.mkString(", "))).map(_.sum)
+  }
+
+  private def getPriceOfCall(call: Call, priceList: PriceList, startTime: Instant, endTime: Instant): Either[String, Double] = {
+    for {
+      _ <- if (call.status.asString == "Success") Right(()) else Left(s"Call {name} status was ${call.status.asString}.") // not evaluating workflows that are in flight or Failed or Aborted or whatever
```

Comment:
why are we not evaluating aborted or failed? They cost money too.

Reply:
Yes! I meant to change that to evaluating all terminal statuses, not just successes